src/Server.cpp

```diff
@@ -200,73 +313,352 @@ void Server::acceptConnection() const
+			return true;
+	}
+	*/
+}
+
+/**
+ * @brief Register a virtual server.
+ *
+ * Adds a virtual server to the list of virtual servers and registers it with epoll.
+ *
+ * @param serverSock The socket of the virtual server to register.
+ *
+ * @return true if the virtual server was successfully registered, false otherwise.
+ */
+bool Server::registerVirtualServer(const Socket& serverSock)
+{
+	struct epoll_event event = {};
+	event.events = EPOLLIN;
+	event.data.fd = serverSock.fd;
+
+	if (!addEvent(m_epfd, serverSock.fd, &event)) {
+		close(serverSock.fd);
+		LOG_ERROR << "Failed to add event for " << serverSock;
+		return false;
+	}
+
+	m_virtualServers[serverSock.fd] = serverSock;
+
+	LOG_INFO << "New virtual server: " << serverSock;
+	return true;
+}
+
+/**
+ * @brief Register a connection.
+ *
+ * Registers a client fd with epoll.
+ * If it fails to add the event, it logs an error and closes the client socket.
+ * If the connection is successfully registered, adds the connection to map m_connections.
+ * The []-operator creates a new entry in the map if the key does not exist or overwrites an existing one.
+ * The same way (re-)initializes the map m_connectionBuffers[newFD] to an empty string.
+ *
+ * @param serverSock The socket of the server that the connection is associated with.
+ * @param clientSock The socket of the client connection to register.
+ *
+ * @return true if the connection was successfully registered, false otherwise.
+ */
+bool Server::registerConnection(const Socket& serverSock, const Socket& clientSock)
+{
+	struct epoll_event event = {};
+	event.events = EPOLLIN;
+	event.data.fd = clientSock.fd;
+
+	if (!addEvent(m_epfd, clientSock.fd, &event)) {
+		close(clientSock.fd);
+		LOG_ERROR << "Failed to add event for " << clientSock;
+		return false;
+	}
+
+	m_connections[clientSock.fd] = Connection(serverSock, clientSock);
+	m_connectionBuffers[clientSock.fd] = "";
+
+	LOG_INFO << "New Connection: " << clientSock << " for server: " << serverSock;
+	return true;
+}
+
+/* ====== HELPER FUNCTIONS ====== */
+
+/**
+ * @brief Create a listening socket.
+ *
+ * Creates a new socket with the provided address information and binds it to the address.
+ * It sets the socket to non-blocking mode and listens for incoming connections.
+ *
+ * @param addrinfo The address information for the socket.
+ * @param backlog The maximum length to which the queue of pending connections may grow.
+ *
+ * @return The file descriptor of the new socket if successful, -1 otherwise.
+ */
+int createListeningSocket(struct addrinfo* addrinfo, int backlog)
+{
+	addrinfo->ai_socktype |= SOCK_NONBLOCK;
+
+	const int newFd = socket(addrinfo->ai_family, addrinfo->ai_socktype, addrinfo->ai_protocol);
+	if (newFd == -1) {
+		LOG_DEBUG << "socket(): " << strerror(errno);
+		return -1;
+	}
+
+	int reuse = 1;
+	if (-1 == setsockopt(newFd, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(int))) {
```

Comment:
Any specific reason why "-1' is before "=="? This scheme gets also used for other return value checks in this file

Reply:
These are called [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions): You swap the variable with the value.

It can protect against unintentional assignments ( 1 = value would error)

But in this case I used them because the C functions are so long with the params, that I misread the condition one time:
`if ((setsockopt(newFd, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(int)) == -1))`

I wanted to make it uniformly in the function createListeningSocket() but maybe somewhere else I did them too :0

Reply:
Quiet interesting, never heard of this before. 
Makes definitely sense for your case and preventing unintentional assignments. 

Yes, you used them also in the following three functions: 

- addEvent
- removeEvent
- modifyEvent